openapi.yaml

```diff
@@ -16,175 +16,511 @@ host: api.oregonstate.edu
-            example: Harry Potter
-          species:
-            type: string
-            description: Species of pet
-            example: Owl
-  PetResult:
+          players:
+            $ref: '#/definitions/players'
+  card:
+    properties:
+      attributes:
+        type: object
+        properties:
+          cardNumber:
+            $ref: '#/definitions/cardNumber'
+          cardSuit:
+            $ref: '#/definitions/cardSuit'
+  cardNumber:
+    type: string
+    enum: 
+      - 'A'
+      - '2'
+      - '3'
+      - '4'
+      - '5'
+      - '6'
+      - '7'
+      - '8'
+      - '9'
+      - '10'
+      - 'J'
+      - 'Q'
+      - 'K'
+  cardSuit:
+    type: string
+    enum:
+      - 'Diamond'
+      - 'Club'
+      - 'Spade'
+      - 'Heart'
+      
+  GamePostBody:
+    properties:
+      data:
+        type: object
+        properties:
+          type:
+            $ref: '#/definitions/gameType'
+          attributes:
+            type: object
+            properties:
+              playerNum:
+                $ref: '#/definitions/playerNum'
+              minimumBet:
+                $ref: '#/definitions/minimumBet'
+              maximumBet:
+                $ref: '#/definitions/maximumBet'
+            required:
+              - playerNum
+              - minimumBet
+              - maximumBet
+            additionalProperties: false
+        required: 
+          - type
+          - attributes
+        additionalProperties: false
+    required:
+      - data
+    additionalProperties: false
+  
+  GameUpdateBody:
+    properties:
+      data:
+        type: object
+        properties:
+          type:
+            $ref: '#/definitions/gameType'
+          attributes:
+            type: object
+            properties:
+              round:
+                $ref: '#/definitions/round'
+              players:
+                $ref: '#/definitions/players'
+              tableCards:
+                $ref: '#/definitions/cards'
+            required:
+              - round
+              - players
+              - tableCards
+            additionalProperties: false
+        required: 
+          - type
+          - attributes
+        additionalProperties: false
+  PlayerUpdateBody:
+    properties:
+      data:
+        type: object
+        properties:
+          type:
+            $ref: '#/definitions/playerType'
+          attributes:
+            type: object
+            properties:
+              player:
+                $ref: '#/definitions/player'
+            required:
+              - player
+            additionalProperties: false
+        required: 
+          - type
+          - attributes
+        additionalProperties: false
+    required: 
+      - data
+    additionalProperties: false
+  gameType:
+    type: string
+    enum:
+      - game
+  playerType:
+    type: string
+    enum:
+      - player
+  playerID:
+    type: string
+    example: 'g1p1'
+    pattern: '^g[1-9]\d*p[1-9]\d*$'
```

Comment:
So here is my question. Should a player be globally unique? If yes, why the `playerId` should start with the `gameId`? If not, why we need to provide a global `GET /players/{playerId}` method? Shouldn't we just make the method under `/games` like this `GET /games/{gameId}/players/{playerId}`?

Reply:
The player is globally unique as you can access its information using gxpx. The player will not be needed after the game.

Reply:
In my design there is no need to use a player twice, but as long as it lives through a lifespan of game, it's still globally unique.